evap/evaluation/tools.py

```diff
@@ -330,3 +330,10 @@ def sort_formset(request, formset):
     if request.POST: # if not, there will be no cleaned_data and the models should already be sorted anyways
         formset.is_valid() # make sure all forms have cleaned_data
         formset.forms.sort(key=lambda f: f.cleaned_data.get("order", 9001) )
+
+def course_types_in_semester(semester):
+    courses = Course.objects.filter(semester=semester)
+    types = set()
+    for course in courses:
+        types.add(course.type)
+    return types
```

Comment:
`return Course.objects.filter(semester=semester).values_list('type', flat=True).distinct()`
or something like that... and then you could inline it. add an `order_by` if necessary.